sound/soc/sof/intel/hda.c

```diff
@@ -101,23 +108,30 @@ int hda_ctrl_dai_widget_free(struct snd_soc_dapm_widget *w, unsigned int quirk_f
 	struct snd_sof_widget *swidget = w->dobj.private;
 	struct snd_soc_component *component = swidget->scomp;
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct sof_dai_private_data *private;
 	struct sof_ipc_dai_config *config;
 	struct snd_sof_dai *sof_dai;
 	struct sof_ipc_reply reply;
 	int ret;
 
 	sof_dai = swidget->private;
 
-	if (!sof_dai || !sof_dai->dai_config) {
-		dev_err(sdev->dev, "error: No config to free DAI %s\n", w->name);
+	if (!sof_dai || !sof_dai->private) {
+		dev_err(sdev->dev, "error: No private data to free DAI %s\n", w->name);
```

Comment:
nitpick: Can we use `dev_err(sdev->dev, "No private data for DAI %s in %s\n", w->name, __func__);` for consistency? The same as `dev_err(sdev->dev, "No config to free DAI %s\n", w->name);` below.